storage/src/test/java/org/apache/kafka/server/log/remote/storage/LocalTieredStorage.java

```diff
@@ -307,7 +307,7 @@ public Optional<CustomMetadata> copyLogSegmentData(final RemoteLogSegmentMetadat
             throws RemoteStorageException {
         Callable<Optional<CustomMetadata>> callable = () -> {
             final RemoteLogSegmentId id = metadata.remoteLogSegmentId();
```

Comment:
On line 316 you can change `id.topicIdPartition()` to `metadata.topicIdPartition()` and remove this line